factom/client.py

```diff
@@ -301,31 +301,58 @@ def transaction(self, tx_hash: Union[bytes, str]):
 
     # Convenience methods
 
-    def read_chain(self, chain_id: Union[bytes, str], include_entry_context: bool = False, encode_as_hex: bool = False):
-        """
-        Shortcut method to read an entire chain.
-
-        Args:
-            chain_id (str): Chain ID to read.
-            include_entry_context (bool): Whether to include information like entry hash, timestamp, and block height
-            encode_as_hex (bool): Whether to encode external-ids and content in entry objects as hex or not
-
-        Returns:
-            list[dict]: A list of entry dictionaries in reverse chronologial order.
-        """
-        entries = []
+    def read_chain(
+        self,
+        chain_id: Union[bytes, str],
+        from_height: int = 0,
+        include_entry_context: bool = False,
+        encode_as_hex: bool = False,
+    ):
+        """A generator that yields all entries of a chain in order, optionally starting from a given block height."""
+        # Walk the entry block chain backwards to build up a stack of entry blocks to fetch
+        entry_blocks = []
         keymr = self.chain_head(chain_id)["chainhead"]
         while keymr != NULL_BLOCK:
             block = self.entry_block(keymr)
-            for entry_pointer in reversed(block["entrylist"]):
+            if block["header"]["dbheight"] < from_height:
```

Comment:
I believe it's correct as is. Because `chain_head()` gets us the most recent eblock, and then we walk he linked list backwards until we find an eblock that has a height less than our start height

Reply:
Indeed you're right, my bad. I thought it's starting from the first block, but clearly the chainhead starts from the most recent one.